src/platform/nrfconnect/CHIPDevicePlatformConfig.h

```diff
@@ -33,8 +36,23 @@
 #define CHIP_DEVICE_CONFIG_DEVICE_PRODUCT_NAME CONFIG_CHIP_DEVICE_PRODUCT_NAME
 #define CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION CONFIG_CHIP_DEVICE_HARDWARE_VERSION
 #define CHIP_DEVICE_CONFIG_DEFAULT_DEVICE_HARDWARE_VERSION_STRING CONFIG_CHIP_DEVICE_HARDWARE_VERSION_STRING
+
+#ifndef CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION
+#ifdef CONFIG_CHIP_DEVICE_SOFTWARE_VERSION
 #define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION CONFIG_CHIP_DEVICE_SOFTWARE_VERSION
+#else
+#define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION APPVERSION
+#endif
+#endif
```

Comment:
```suggestion
#ifdef CONFIG_CHIP_DEVICE_SOFTWARE_VERSION
#define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION CONFIG_CHIP_DEVICE_SOFTWARE_VERSION
#else
#define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION APPVERSION
#endif
```

IMHO, if the corresponding Kconfig option is available it should always be used to populate the given Matter's config. Otherwise, if a user forgets to remove the manual `#define`, the code will use a different version than the OTA image and no error/warning will be raised.